client/src/Pages/DashboardProvider/index.js

```diff
@@ -49,8 +87,14 @@ const DashboardProvider = () => {
 
   const availability = async (checked) => {
     try {
-      await Axios.patch('/api/v1/provider/availability');
-      console.log(checked);
+      setSwitchLoading(true);
+      await Axios.patch('/api/v1/provider/availability', {
+        id: providerDetails?.id,
```

Comment:
why you wait for another request result to get the id value when you have the id on your context ``` userData.id ```
 + the same issue here ``` no need to send param, it has it on middleware ```